Docs/source/developers/testing.rst

```diff
@@ -47,97 +30,149 @@ You can run those locally before you push to save some time, install them once l
-However, when running the tests on a local machine, it is usually possible and convenient to allot more threads for compiling, in order to speed up the builds.
-This can be accomplished by setting the environment variable ``WARPX_CI_NUM_MAKE_JOBS``, with the preferred number of threads that fits your local machine, e.g. ``export WARPX_CI_NUM_MAKE_JOBS=16`` (or less if your machine is smaller).
-On public CI, we overwrite the value to ``WARPX_CI_NUM_MAKE_JOBS=2``, in order to avoid overloading the available remote resources.
-Note that this will not change the number of threads used to run each test, but only the number of threads used to compile each executable necessary to run the tests.
+  .. code-block:: sh
 
-Once the execution of `./run_test.sh <https://github.com/ECP-WarpX/WarpX/blob/development/run_test.sh>`__ is completed, you can find all the relevant files associated with each test in one single directory.
-For example, if you run the single test ``pml_x_yee``, as shown above, on 04/30/2021,  you can find all relevant files in the directory ``./test_dir/rt-WarpX/WarpX-tests/2021-04-30/pml_x_yee/``.
-The content of this directory will look like the following (possibly including backtraces if the test crashed at runtime):
+       add_warpx_test(
+           test_2d_laser_acceleration_picmi # name
+           2 # dims
+           2 # nprocs
+           OFF # eb
+           Examples/Tests/laser_acceleration/inputs_test_2d_laser_acceleration_picmi.py # inputs
+           Examples/Tests/laser_acceleration/analysis.py # analysis
+           diags/diag1000100 # output (plotfile)
+           OFF # dependency
+       )
 
-.. code-block:: sh
+* Add the **restart test** ``test_3d_laser_acceleration_restart``, with input file in ``Examples/Tests/laser_acceleration/`` and default restart analysis script provided by WarpX:
+
+  .. code-block:: sh
 
-    $ ls ./test_dir/rt-WarpX/WarpX-tests/2021-04-30/pml_x_yee/
-    analysis_pml_yee.py     # Python analysis script
-    inputs_2d               # input file
-    main2d.gnu.TEST.TPROF.MTMPI.OMP.QED.ex  # executable
-    pml_x_yee.analysis.out  # Python analysis output
-    pml_x_yee.err.out       # error output
-    pml_x_yee.make.out      # build output
-    pml_x_yee_plt00000/     # data output (initialization)
-    pml_x_yee_plt00300/     # data output (last time step)
-    pml_x_yee.run.out       # test output
+       add_warpx_test(
+           test_3d_laser_acceleration_restart # name
+           3 # dims
+           2 # nprocs
+           OFF # eb
+           Examples/Tests/laser_acceleration/inputs_test_3d_laser_acceleration_restart # inputs
+           Examples/analysis_default_restart.py # analysis
+           diags/diag1000100 # output (plotfile)
+           test_3d_laser_acceleration # dependency
```

Comment:
```suggestion
           test_3d_laser_acceleration_restart  # name
           3  # dims
           2  # nprocs
           OFF  # eb
           Examples/Tests/laser_acceleration/inputs_test_3d_laser_acceleration_restart  # inputs
           Examples/analysis_default_restart.py  # analysis
           diags/diag1000100  # output (plotfile)
           test_3d_laser_acceleration  # dependency
```

Reply:
Done in [8e64870](https://github.com/ECP-WarpX/WarpX/pull/5068/commits/8e64870fe4fe42fd6c88f69741fe7761d30226c6).